src/split.cpp

```diff
@@ -118,3 +136,54 @@ List findSplit(const NumericVector x, const IntegerVector y, const int & ndSize,
 	}
 	return List::create(_["MaxDeltaI"] = maxdI, _["BestVar"] = bv, _["BestSplit"] = bs);
 }
+
+double mean(NumericVector x) {
+    int n = x.size();
+    double total = 0;
+
+    for(int i = 0; i < n; ++i) {
+        total += x[i];
+    }
+    return total / n;
+}
+
+double mse(NumericVector y) {
+    double y_mean = mean(y);
+    return sum(pow((y_mean - y),2));
+}
+
+
+List findSplitRegression(const NumericVector x, const NumericVector y, const NumericVector splitPoints, const int & ndSize, const double & I,
+	       double maxdI, int bv, double bs, const int nzidx) {
+    double dI;
+    int splitN = splitPoints.size();
+    int splitEnd;
+
+    if (splitN > 1) {
+        for (int i = 1; i < splitN; ++i) {
+            splitEnd = splitPoints[i]-2;
+            dI = I - sum(NumericVector::create(mse(y[Range(0,splitEnd)]), mse(y[Range(splitEnd+1,ndSize-1)])));
+            if (dI > maxdI) {
+                maxdI = dI;
+                bv = nzidx;
+                bs = ((double)x[splitEnd] + (double)x[splitEnd+1]) / 2.0;
+            }
+        }
+    }
+	return List::create(_["MaxDeltaI"] = maxdI, _["BestVar"] = bv, _["BestSplit"] = bs);
+}
+
+
+// [[Rcpp::export]]
+List findSplit(const NumericVector x, const NumericVector y, const NumericVector splitPoints, const int & ndSize, const double & I,
+	       double maxdI, int bv, double bs, const int nzidx, arma::vec cc, const int & task) {
+
+    List ret;
+    if (task == 0) {
```

Comment:
can we be more explicit about what we are doing here?

Reply:
Added comments.